### ISSUE TITLE
Take into account final modifiers for start and fixed attributes in parameter input windows

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#12898]
Consider the following MWE:
```modelica
package TestFinalModifiers
  model M
    parameter Real x_start = 0;
    Real x1(final start = x_start, final fixed = true);
    Real x2(final start = x_start);
    Real x3(final fixed = true);
  equation
    der(x1) = -x1;
    der(x2) = -x2;
    der(x3) = -x3;
  end M;

  model S
    M m annotation(
      Placement(transformation(origin = {-4, 4}, extent = {{-10, -10}, {10, 10}})));
  end S;
end TestFinalModifiers;
```
When I double-click on `m` in `S` I get this window:
![immagine](https://github.com/user-attachments/assets/a511815e-01ca-491c-b61b-56f2899bf965)

which allows me to change the values of the start and fixed attributes for all the three variables x1, x2, and x3. This is confusing for a user, because it seems there are two ways to change the start attribute, whereas there should be only one. 

Additionally, if the input fields for the start and fixed attributes are actually used, invalid models are generated which attempt to modify final attributes.

Bottom line:
- the x1.start line should not be shown at all in the Initialization tab
- the x2.start line should only show the fixed checkbox
- the x3.start line should only show the start value input field

**Comment by casella** [comment]
Thank you @adeas31!

@ceraolo, as soon as we have PowerGrids ready, we can make a 1.24.1 release with this feature, if it hasn't been done already for other reasons.

**Comment by ceraolo** [comment]
> @ceraolo, as soon as we have PowerGrids ready, we can make a 1.24.1 release with this feature, if it hasn't been done already for other reasons.

Excellent!

**Comment by AndreaBartolini** [comment]
Some problems are still present, I checked the MWE above and by double clicking on the model `m` the following pop-up is shown:

<img width="584" height="471" alt="Image" src="https://github.com/user-attachments/assets/267ce6d2-5f36-4fcc-99e7-fe7142f3dc63" />

for the attribute `x2.start` the possibility to set the `fixed` modifier is shown and this is correct because that modifier is not set in the declaration of `x2`

```modelica
  Real x2(final start = x_start);
```

for the variable `x3` nothing it shown but the `start` attribute is not set in its declaration

```modelica
  Real x3(final fixed = true);
```
so that `start` attribute can be set only in the _text_ view or, after the first simulation, in the _plotting_ view (as shown below)

<img width="223" height="178" alt="Image" src="https://github.com/user-attachments/assets/9dc4620d-1215-4f4d-8696-00c0fbad181c" />

The expected behavior is that in the pop-up the user can set the value of the `x3.start` but not the `x3.fixed` modifier.

sysop Ubuntu 24.04 under Win11-pro WSL
OpenModelica 1.28.0~dev-226-gb650b2f

**Comment by adeas31** [comment]
@AndreaBartolini this is intentional. We only list the variables that have either `start` or `fixed` attribute defined. For `Real x3(final fixed = true);`, the `start` is not defined and `fixed` is marked as `final` so nothing is shown.

If we list such variables then we need to list all variables like `Real x4; Real foo;` etc. which will clutter the parameters window for large models.

**Comment by AndreaBartolini** [comment]
Maybe I've not completely understood the key point of this ticket... we need to properly set the `start` and (eventually) the `fixed` attributes only for the potential _state_ variables, not for all the variables.

In my opinion the logic should applied only to the variables that are argument of `der()` operator or are set to be _potential_ state by using the attribute `stateSelect`.

For said variables are very important to set the `start` value so to remove them from the pop-up when the `fixed` attribute is set as `final` is not a good idea from my point of view...

**Comment by casella** [comment]
@AndreaBartolini, as @adeas31 wrote, we are not putting v.start fields for _all_ variables v in a model, because they would be too many. Since version 1.27.0 you can set `x3.start` and `x3.nominal` from the Modifiers tab.

If you want to see `x3.start` in the General tab below `x3.start`, please add an _explicit_ modifier to `x3.start` to the code, i.e.
```
Real x3(final fixed = true, start = 0);
```

**Comment by casella** [comment]
@AndreaBartolini you suggested

> In my opinion the logic should applied only to the variables that are argument of der() operator or are set to be potential state by using the attribute stateSelect.

This would be problematic, state selection is handled by the backend, whereas OMEdit gets information from the frontend only (backend would be too slow).

The idea is that the modeller knows which values may need a start modifier; he/she needs to add one to the code, then OMEdit reads it and puts it in the General tab, Initialization group.
